src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -67,6 +67,10 @@ public function denormalize($data, $class, $format = null, array $context = arra
     {
         $dateTimeFormat = isset($context[self::FORMAT_KEY]) ? $context[self::FORMAT_KEY] : null;
 
+        if ('' === $data || null === $data) {
+            throw new UnexpectedValueException('The data is either an empty string or null, you should pass an object implementing \DateTimeInterface.');
```

Comment:
This message is wrong: we don't expect an object implementing `\DateTimeInterface`. We expect a string that can be parsed according to the configured format or the `\DateTime::__construct()` if not configured.